### PR TITLE
Add custom label prefix support for cherry picker

### PR DESCRIPTION
**Pull request by hi-rustin**
We want to use a custom prefix to be consistent with the label style we currently use.

I think other communities also have needs. Maybe we can migrate these labels, but I think it is more reasonable to support custom label style.

### CODE REVIEW

File: prow/external-plugins/cherrypicker/lib/lib.go [hunk 1]
```diff
@@ -19,10 +19,10 @@ package cherrypicker
 import "fmt"
 
 // CreateCherrypickBody creates the body of a cherrypick PR
-func CreateCherrypickBody(num int, requestor, note string) string {
+func CreateCherrypickBody(num int, requester, note string) string {
 	cherryPickBody := fmt.Sprintf("This is an automated cherry-pick of #%d", num)
-	if len(requestor) != 0 {
-		cherryPickBody = fmt.Sprintf("%s\n\n/assign %s", cherryPickBody, requestor)
+	if len(requester) != 0 {
+		cherryPickBody = fmt.Sprintf("%s\n\n/assign %s", cherryPickBody, requester)
```
Comment (petr-muller) [inline prow/external-plugins/cherrypicker/lib/lib.go:25]: Please remove these, `requestor` is a valid variant of the [word](https://dictionary.cambridge.org/dictionary/english/requester) so this just pollutes the change.

/ok-to-test

**Review by petr-muller — CHANGES_REQUESTED** [review]

**Comment by petr-muller** [comment]
What's the rationale? Why should cherrypick-related labels have a different prefix?

**Comment by hi-rustin** [comment]
> What's the rationale? Why should cherrypick-related labels have a different prefix?

Because we want to use a custom prefix to be consistent with the label style we currently use.

I think other communities also have needs. Maybe we can migrate these labels, but I think it is a more reasonable way to support custom label style.

File: prow/external-plugins/cherrypicker/server_test.go [hunk 2]
```diff
@@ -570,13 +580,13 @@ func testCherryPickPRWithLabels(clients localgit.Clients, t *testing.T) {
 			},
 			prLabels: []github.Label{
 				{
-					Name: "cherrypick/release-1.5",
+					Name: labelPrefix + "release-1.5",
```
Comment (stevekuznetsov) [inline prow/external-plugins/cherrypicker/server_test.go:583]: nit: Prefer not to parameterize your test cases. This test is not only testing the output of the plugin but also testing that you correctly create test data here at runtime. Passing static data would be preferred.

File: prow/external-plugins/cherrypicker/main.go [hunk 3]
```diff
@@ -68,10 +69,11 @@ func gatherOptions() options {
 	fs.BoolVar(&o.prowAssignments, "use-prow-assignments", true, "Use prow commands to assign cherrypicked PRs.")
 	fs.BoolVar(&o.allowAll, "allow-all", false, "Allow anybody to use automated cherrypicks by skipping GitHub organization membership checks.")
 	fs.BoolVar(&o.issueOnConflict, "create-issue-on-conflict", false, "Create a GitHub issue and assign it to the requestor on cherrypick conflict.")
+	fs.StringVar(&o.labelPrefix, "label-prefix", "", "Set a custom label prefix.")
```
Comment (stevekuznetsov) [inline prow/external-plugins/cherrypicker/main.go:72]: You could expose the default prefix as a `const` and make it the default value of this flag, so you don't need the `len() != 0` defaulting logic later.

Comment (hi-rustin) [reply prow/external-plugins/cherrypicker/main.go]: Fixed.

File: prow/external-plugins/cherrypicker/server_test.go (hunk 2 above)
Comment (hi-rustin) [reply prow/external-plugins/cherrypicker/server_test.go]: Fixed.

**Comment by hi-rustin** [comment]
@stevekuznetsov  How about now?
PTAL~ Thanks!

**Comment by hi-rustin** [comment]
@stevekuznetsov ping~

**Review by kargakis — APPROVED** [review]

**Pull request merged**